spec/integration/operations_spec.rb

```diff
@@ -86,6 +86,62 @@ module Hawkular::Operations::RSpec
       end
     end
 
+    it 'should run into error callback because bad hash parameters' do
+      WebSocketVCR.configure do |c|
+        c.hook_uris = [HOST]
+      end
+
+      WebSocketVCR.record(example, self) do
+        client = OperationsClient.new(host: HOST,
+                                      wait_time: WebSocketVCR.live? ? 1.5 : 2,
+                                      credentials: {
+                                        username: 'jdoe',
+                                        password: 'password'
+                                      },
+                                      options: {
+                                        tenant: 'hawkular'
+                                      })
+
+        noop = { operationName: 'noop' }
+
+        client.invoke_generic_operation(noop) do |on|
+          on.success do |_data|
+            fail 'This should have failed'
+          end
+          on.failure do |error|
+            puts 'error callback was correctly called, reason: ' + error.to_s
+          end
+        end
+      end
+    end
+
+    it 'should bail with hash property error' do
+      WebSocketVCR.configure do |c|
+        c.hook_uris = [HOST]
+      end
+
+      WebSocketVCR.record(example, self) do
+        client = OperationsClient.new(host: HOST,
+                                      wait_time: WebSocketVCR.live? ? 1.5 : 2,
+                                      credentials: {
+                                        username: 'jdoe',
+                                        password: 'password'
+                                      },
+                                      options: {
+                                        tenant: 'hawkular'
+                                      })
+
+        noop = { operationName: 'noop' }
+        begin
+          client.invoke_generic_operation(noop)
+        rescue
+          puts 'We got an exception and this is good'
+        else
+          fail 'Should have failed with error:Hash property must be specified'
+        end
```

Comment:
You could use something like:
expect { client.invoke_generic_operation(noop) }.to raise_error(ArgumentError)